File: user-service/src/main/java/com/bank/userservice/service/kafka/sender/KafkaMetricSender.java
package com.bank.userservice.service.kafka.sender;

import com.bank.userservice.dto.kafka.MetricDto;
import com.bank.userservice.service.kafka.KafkaAsyncSender;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

@Component
public class KafkaMetricSender extends KafkaAsyncSender<MetricDto> {

    public KafkaMetricSender(KafkaTemplate<String, Object> template,
                             ObjectMapper objectMapper) {
        super(template, objectMapper, "metrics-topic");
    }
}

File: core-query/src/main/kotlin/ru/patterns/corequery/utils/metric/KafkaMetricSender.kt
package ru.patterns.corequery.utils.metric

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.stereotype.Component
import reactor.kafka.sender.KafkaSender
import ru.patterns.corequery.utils.KafkaAsyncSender
import ru.patterns.corequery.utils.metric.serialization.MetricDto

@Component
class KafkaMetricSender(
    kafkaSender: KafkaSender<String, String>,
    objectMapper: ObjectMapper,
) : KafkaAsyncSender<MetricDto>(
    kafkaSender = kafkaSender,
    objectMapper = objectMapper,
    topic = "metrics-topic"
)